node/pkg/accounting/accounting.go

```diff
@@ -0,0 +1,353 @@
+	acct.logger.Debug("acct: leaving AuditPendingTransfers")
+}
+
+// publishTransfer publishes a pending transfer to the accounting channel and updates the timestamp. It assumes the caller holds the lock.
+func (acct *Accounting) publishTransfer(pe *pendingEntry) {
+	if acct.enforceFlag {
+		acct.logger.Debug("acct: publishTransfer: notifying the processor", zap.String("msgId", pe.msgId))
+		acct.msgChan <- pe.msg
+	}
+
+	acct.deletePendingTransfer(pe.msgId)
+}
+
+// addPendingTransfer adds a pending transfer to both the map and the database. It assumes the caller holds the lock.
+func (acct *Accounting) addPendingTransfer(msgId string, msg *common.MessagePublication, digest string) error {
+	acct.logger.Debug("acct: addPendingTransfer", zap.String("msgId", msgId))
+	if err := acct.db.AcctStorePendingTransfer(msg); err != nil {
+		return err
+	}
+
+	pe := &pendingEntry{msg: msg, msgId: msgId, digest: digest, updTime: time.Now()}
+	acct.pendingTransfers[msgId] = pe
+	transfersOutstanding.Inc()
+	return nil
+}
+
+// deletePendingTransfer deletes the transfer from both the map and the database. It assumes the caller holds the lock.
+func (acct *Accounting) deletePendingTransfer(msgId string) {
+	acct.logger.Debug("acct: deletePendingTransfer", zap.String("msgId", msgId))
+	if _, exists := acct.pendingTransfers[msgId]; exists {
+		transfersOutstanding.Dec()
+		delete(acct.pendingTransfers, msgId)
+	}
+	if err := acct.db.AcctDeletePendingTransfer(msgId); err != nil {
+		acct.logger.Error("acct: failed to delete pending transfer from the db", zap.String("msgId", msgId), zap.Error(err))
+		// Ignore this error and keep going.
+	}
+}
+
+// loadPendingTransfers loads any pending transfers that are present in the database. This method assumes the caller holds the lock.
+func (acct *Accounting) loadPendingTransfers() error {
+	pendingTransfers, err := acct.db.AcctGetData(acct.logger)
+	if err != nil {
+		return err
+	}
+
+	for _, msg := range pendingTransfers {
+		msgId := msg.MessageIDString()
+		acct.logger.Info("acct: reloaded pending transfer", zap.String("msgID", msgId))
+
+		digest := msg.CreateDigest()
+		pe := &pendingEntry{msg: msg, msgId: msgId, digest: digest} // Leave the updTime unset so we will query this on the first audit interval.
+		acct.pendingTransfers[msgId] = pe
+		transfersOutstanding.Inc()
+	}
+
+	if len(acct.pendingTransfers) != 0 {
+		acct.logger.Info("acct: reloaded pending transfers", zap.Int("total", len(acct.pendingTransfers)))
+	} else {
+		acct.logger.Info("acct: no pending transfers to be reloaded")
+	}
+
+	return nil
+}
+
+// submitObservation sends an observation request to the worker so it can be submited to the contract.
+// If writing to the channel would block, this function resets the timestamp on the entry so it will be
+// retried next audit interval. This method assumes the caller holds the lock.
+func (acct *Accounting) submitObservation(msg *common.MessagePublication) {
+	if len(acct.subChan) < cap(acct.subChan) {
```

Comment:
This is racy.  The idiomatic way to try sending on a channel is to use a select statement:

```go
select {
	case acct.subChan <- msg:
		// Success
	default:
		// Handle send failure
}
```

Reply:
Fixed.